e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1373,8 +1373,8 @@ func ResourceGroupStatusEquals(expected v1alpha1.ResourceGroupStatus) Predicate
 			}
 		}
 		if !equality.Semantic.DeepEqual(found, expected) {
-			return fmt.Errorf("ResourceGroup %s status does not match expected value:\nDiff (- Expected, + Found)\n%s",
-				client.ObjectKeyFromObject(rg), log.AsYAMLDiff(expected, found))
+			return fmt.Errorf("ResourceGroup %s status does not match expected value:\nExpected:\n%s\nFound:\n%s",
+				client.ObjectKeyFromObject(rg), log.AsYAML(expected), log.AsYAML(found))
```

Comment:
I'd think printing the entire yaml of both objects would be just as verbose, if not more verbose than printing the diff between the objects. Am I missing something?

Reply:
The problem isn't the string size in the log or even serializing objects to yaml. The problem is the diff algorithm's memory usage for large strings with lots of diffs. Basically every object that isn't Current has multiple diff lines and a large-ish conditions field.